crates/gitbutler-core/src/virtual_branches/integration.rs

```diff
@@ -404,3 +412,29 @@ fn verify_head_is_set(
         Some(head_name) => Err(errors::VerifyError::InvalidHead(head_name.to_string())),
     }
 }
+
+fn is_integration_commit(commit: &git::Commit) -> bool {
+    is_integration_commit_author(commit) && is_integration_commit_message(commit)
+}
+
+fn is_integration_commit_author(commit: &git::Commit) -> bool {
+    is_integration_commit_author_email(commit) && is_integration_commit_author_name(commit)
+}
+
+fn is_integration_commit_author_email(commit: &git::Commit) -> bool {
+    commit.author().email().map_or(false, |email| {
+        email == GITBUTLER_INTEGRATION_COMMIT_AUTHOR_EMAIL
+    })
+}
+
+fn is_integration_commit_author_name(commit: &git::Commit) -> bool {
+    commit.author().name().map_or(false, |name| {
+        name == GITBUTLER_INTEGRATION_COMMIT_AUTHOR_NAME
+    })
+}
+
+fn is_integration_commit_message(commit: &git::Commit) -> bool {
+    commit
+        .message()
+        .starts_with(b"GitButler Integration Commit")
+}
```

Comment:
What do you think about collapsing this into just one function? Feels like that would be easier to read.